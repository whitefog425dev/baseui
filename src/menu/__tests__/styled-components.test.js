/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import * as React from 'react';
import {shallow} from 'enzyme';
import {
  List,
  ListItem,
  ListItemProfile,
  ProfileBody,
  ProfileImg,
  ProfileTitle,
  ProfileSubtitle,
  ProfileImgContainer,
  ProfileLabelsContainer,
} from '../styled-components';

function makeTest({
  title,
  component: Component,
  props = {},
  children = null,
  snapshotName = 'correct styles',
}: {
  title: string,
  component: React.ComponentType<*>,
  props?: {},
  children?: *,
  snapshotName?: string,
}) {
  test(title, () => {
    const shallowed = shallow(<Component {...props}>{children}</Component>);
    expect(shallowed.instance().getStyles()).toMatchSnapshot(snapshotName);
  });
}

describe('Menu Styled Components', () => {
  makeTest({
    title: 'ListItem - basic render',
    component: ListItem,
  });

  makeTest({
    title: 'ListItem - highlighted render',
    component: ListItem,
    props: {
      $isHighlighted: true,
    },
  });

  makeTest({
    title: 'List - basic render',
    component: List,
  });

  makeTest({
    title: 'ListItemProfile - basic render',
    component: ListItemProfile,
  });

  makeTest({
    title: 'ProfileImgContainer - basic render',
    component: ProfileImgContainer,
  });

  makeTest({
    title: 'ProfileImg - basic render',
    component: ProfileImg,
  });

  makeTest({
    title: 'ProfileLabelsContainer - basic render',
    component: ProfileLabelsContainer,
  });

  makeTest({
    title: 'ProfileTitle - basic render',
    component: ProfileTitle,
  });

  makeTest({
    title: 'ProfileSubtitle - basic render',
    component: ProfileSubtitle,
  });

  makeTest({
    title: 'ProfileBody - basic render',
    component: ProfileBody,
  });
});
