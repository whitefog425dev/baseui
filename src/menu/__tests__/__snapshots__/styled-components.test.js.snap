// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Styled Components List - basic render: correct styles 1`] = `
Object {
  "background": "$theme.colors.white",
  "borderRadius": "$theme.borders.radius300",
  "boxShadow": "$theme.lighting.shadow600",
  "margin": 0,
  "padding": 0,
  "position": "relative",
}
`;

exports[`Menu Styled Components ListItem - basic render: correct styles 1`] = `
Object {
  ":first-child": Object {
    "paddingTop": "$theme.sizing.scale600",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono300",
  },
  ":last-child": Object {
    "paddingBottom": "$theme.sizing.scale600",
  },
  "color": "$theme.colors.black",
  "cursor": "pointer",
  "display": "block",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "color, background-color",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
}
`;

exports[`Menu Styled Components ListItem - highlighted render: correct styles 1`] = `
Object {
  ":first-child": Object {
    "paddingTop": "$theme.sizing.scale600",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono300",
  },
  ":last-child": Object {
    "paddingBottom": "$theme.sizing.scale600",
  },
  "color": "$theme.colors.primary",
  "cursor": "pointer",
  "display": "block",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "color, background-color",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
}
`;

exports[`Menu Styled Components ListItemProfile - basic render: correct styles 1`] = `
Object {
  ":first-child": Object {
    "paddingTop": "$theme.sizing.scale800",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono300",
  },
  ":last-child": Object {
    "paddingBottom": "$theme.sizing.scale800",
  },
  "alignItems": "center",
  "cursor": "pointer",
  "display": "flex",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "color, background-color",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
}
`;

exports[`Menu Styled Components ProfileBody - basic render: correct styles 1`] = `
Object {
  "color": "$theme.colors.mono800",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "margin": 0,
}
`;

exports[`Menu Styled Components ProfileImg - basic render: correct styles 1`] = `
Object {
  "borderRadius": "50%",
  "height": "100%",
  "width": "100%",
}
`;

exports[`Menu Styled Components ProfileImgContainer - basic render: correct styles 1`] = `
Object {
  "alignItems": "center",
  "display": "flex",
  "height": "60px",
  "justifyContent": "center",
  "width": "60px",
}
`;

exports[`Menu Styled Components ProfileLabelsContainer - basic render: correct styles 1`] = `
Object {
  "alignSelf": "flex-start",
  "display": "flex",
  "flexDirection": "column",
  "marginLeft": "$theme.sizing.scale600",
}
`;

exports[`Menu Styled Components ProfileSubtitle - basic render: correct styles 1`] = `
Object {
  "color": "$theme.colors.black",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "margin": 0,
}
`;

exports[`Menu Styled Components ProfileTitle - basic render: correct styles 1`] = `
Object {
  "color": "$theme.colors.black",
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
  "margin": 0,
}
`;
